exporter/exporterparser/zipkin.go

```diff
@@ -106,89 +110,250 @@ func ZipkinExportersFromYAML(config []byte) (tes []exporter.TraceExporter, doneF
 	return
 }
 
-func newZipkinExporter(finalEndpointURI, defaultServiceName, defaultLocalEndpointURI string) (*zipkinExporter, error) {
-	localEndpoint, err := openzipkin.NewEndpoint(defaultServiceName, defaultLocalEndpointURI)
-	if err != nil {
-		return nil, err
+func newZipkinExporter(finalEndpointURI, defaultServiceName, defaultLocalEndpointURI string, uploadPeriod time.Duration) (*zipkinExporter, error) {
+	var opts []zipkinhttp.ReporterOption
+	if uploadPeriod > 0 {
+		opts = append(opts, zipkinhttp.BatchInterval(uploadPeriod))
 	}
-	reporter := http.NewReporter(finalEndpointURI)
-	defaultExporter := zipkin.NewExporter(reporter, localEndpoint)
+	reporter := zipkinhttp.NewReporter(finalEndpointURI, opts...)
 	zle := &zipkinExporter{
 		endpointURI:             finalEndpointURI,
-		defaultExporter:         defaultExporter,
 		defaultServiceName:      defaultServiceName,
 		defaultLocalEndpointURI: defaultLocalEndpointURI,
-		serviceNameToExporter:   make(map[string]*zipkin.Exporter),
+		reporter:                reporter,
 	}
+	return zle, nil
+}
 
-	// Ensure that we add the default reporter's Close functions
-	zle.shutdownFns = append(zle.shutdownFns, reporter.Close)
+func zipkinEndpointFromNode(node *commonpb.Node, serviceName string, prefixFunc func(string) string) (*zipkinmodel.Endpoint, error) {
+	if node == nil {
+		return nil, nil
+	}
 
-	return zle, nil
+	// The data in the Attributes map was saved in the format
+	// {
+	//      prefixFunc("ipv4"): "192.168.99.101",
+	//      prefixFunc("port"): "9000",
+	//      prefixFunc("serviceName"): "backend",
+	// }
+	attributes := node.Attributes
+
+	var endpointURI string
+
+	ipv6Selected := false
+	if ipv4 := attributes[prefixFunc("ipv4")]; ipv4 != "" {
+		endpointURI = ipv4
+	} else if ipv6 := attributes[prefixFunc("ipv6")]; ipv6 != "" {
+		// In this case for ipv6 address when combining with a port,
+		// we need them enclosed in square brackets as per
+		//    https://golang.org/pkg/net/#SplitHostPort
+		endpointURI = fmt.Sprintf("[%s]", ipv6)
+		ipv6Selected = true
+	}
+
+	port := attributes[prefixFunc("port")]
+	if port != "" {
+		endpointURI += ":" + port
+	} else if ipv6Selected { // net.SplitHostPort requires a port of ipv6 values
```

Comment:
I would change this to:
```go
if port := attributes[prefixFunc("port")]; port == "" {
  endpointURI += ":0"
} else {
  endpointURI += ":"+port
}
return openzipkin.NewEndpoint(serviceName, endpointURI)
```
then you can also drop the ipv6Selected variable above as it is not needed anymore

Reply:
Not really, even for ipv4 we need that port.